### PR TITLE
Add io.github.amit9838.weather

### PR DESCRIPTION
**Pull request by amit9838**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by amit9838** [comment]
bot, build com.github.amit9838.weather

File: com.github.amit9838.weather.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{
+    "app-id" : "com.github.amit9838.weather",
```
Comment (hfiguiere) [inline com.github.amit9838.weather.json:2]: id must be `io.github` as per the documentation:

https://docs.flatpak.org/en/latest/conventions.html#application-ids

File: com.github.amit9838.weather.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        "/man",
+        "/share/doc",
+        "/share/gtk-doc",
+        "/share/man",
+        "/share/pkgconfig",
+        "*.la",
+        "*.a"
+    ],
+    "modules" : [
+        "data/dependencies/python3-requests.json",
```
Comment (hfiguiere) [inline com.github.amit9838.weather.json:26]: this file is no present here.

File: com.github.amit9838.weather.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                    "tag"  : "v0.1.0",
+                    "commit" : "354c722d90b171eb2a6da8670a40f9ca531d7d80",
+                    "url" : "https://github.com/amit9838/weather.git"
+                }
+            ]
+        }
+    ],
+    "build-options" : {
+        "env" : {        }
+    }
```
Comment (hfiguiere) [inline com.github.amit9838.weather.json:43]:
```suggestion
    ]
```

File: flathub.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+    "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: why?

Comment (amit9838) [reply flathub.json:2]: didn't tested for other platforms

Comment (hfiguiere) [reply flathub.json:2]: it's clear the python code will not be tested that way.

**Comment by amit9838** [comment]
bot, build com.github.amit9838.weather

File: com.github.amit9838.weather.json [hunk 5]
```diff
@@ -0,0 +1,41 @@
+{
+    "app-id" : "io.github.amit9838.weather",
```
Comment (hfiguiere) [inline com.github.amit9838.weather.json:2]: the manifest filename MUST match the app-id.

File: com.github.amit9838.weather.json (hunk 2 above)
Comment (hfiguiere) [reply com.github.amit9838.weather.json:26]: you didn't solve this.

Comment (amit9838) [reply com.github.amit9838.weather.json]: can pls, you check it now

**Comment by amit9838** [comment]
bot, build io.github.amit9838.weather

**Comment by amit9838** [comment]
bot, build io.github.amit9838.weather

**Comment by amit9838** [comment]
Can review this now?

**Comment by amit9838** [comment]
Can review now?

**Comment by amit9838** [comment]
Can you pls review?

**Comment by hfiguiere** [comment]
can you please be patient?

**Comment by barthalion** [comment]
```
~ 2.2s ❱ flatpak run io.github.amit9838.weather
Traceback (most recent call last):
  File "/app/bin/weather", line 45, in <module>
    from weather import main
  File "/app/share/weather/weather/main.py", line 28, in <module>
    from .weather import WeatherWindow
  File "/app/share/weather/weather/weather.py", line 8, in <module>
    from .windows import AboutWindow,WeatherPreferences
  File "/app/share/weather/weather/windows.py", line 7, in <module>
    from .backend_current_w import fetch_city_info
  File "/app/share/weather/weather/backend_current_w.py", line 1, in <module>
    import requests
ModuleNotFoundError: No module named 'requests'
```

**Comment by amit9838** [comment]
Added missing dependencies.

**Comment by amit9838** [comment]
bot, build io.github.amit9838.weather

**Comment by amit9838** [comment]
bot, build io.github.amit9838.weather

**Comment by amit9838** [comment]
bot, build io.github.amit9838.weather

**Comment by amit9838** [comment]
Fixed issues, everything is working fine now.

**Pull request closed without merging**